server/src/main/java/org/candlepin/resteasy/DateFormatter.java

```diff
@@ -15,34 +15,74 @@
 package org.candlepin.resteasy;
 
 import org.apache.commons.lang3.StringUtils;
+
 import org.jboss.resteasy.spi.StringParameterUnmarshaller;
 import org.jboss.resteasy.util.FindAnnotation;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.lang.annotation.Annotation;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 
+import javax.xml.bind.DatatypeConverter;
+
+
+
 /**
- * Formatter for dates on REST API parameters
+ * Formatter for parsing dates for parameters annotated with the DateFormat annotation.
  */
 public class DateFormatter implements StringParameterUnmarshaller<Date> {
-    private SimpleDateFormat formatter;
+    private static Logger log = LoggerFactory.getLogger(DateFormatter.class);
+
+    private String[] formats;
 
     public void setAnnotations(Annotation[] annotations) {
-        DateFormat format = FindAnnotation.findAnnotation(annotations, DateFormat.class);
-        formatter = new SimpleDateFormat(format.value());
+        DateFormat annotation = FindAnnotation.findAnnotation(annotations, DateFormat.class);
+
+        this.formats = annotation.value();
     }
 
-    public Date fromString(String str) {
-        if (StringUtils.isBlank(str)) {
-            return new Date(0);
-        }
-        try {
-            return formatter.parse(str);
-        }
-        catch (ParseException e) {
-            throw new RuntimeException(e);
+    public Date fromString(String value) {
+        // If we're null/empty, we have nothing to parse, so return null
+        if (!StringUtils.isBlank(value)) {
+            // If the value is our special value for "now", return a new Date instance
+            if (DateFormat.NOW.equals(value)) {
+                return new Date();
+            }
+
+            // If we have any formats specified, use those...
+            if (this.formats != null && this.formats.length > 0) {
+                for (String format : this.formats) {
+                    try {
+                        SimpleDateFormat formatter = new SimpleDateFormat(format);
+                        return formatter.parse(value);
+                    }
+                    catch (ParseException exception) {
+                        // Whoops. Hopefully we have more formats to try...
+                    }
+                }
+
+                // If we made it here, we're out of formats to use to parse the date
+
+                // TODO: How do we get an i18n instance at this point to translate this?
+                String message = "Unable to parse date parameter: " + value;
+                throw new RuntimeException(message);
+            }
+            else {
+                try {
```

Comment:
@awood Not opposed to that idea, but the difficulty there is we would need to specify all of them in an array and the combine them with the others we want, which wouldn't be usable in the annotation. If it were just one format, or the formatter backing this supported optional bits, that'd be a nice addition.

Also, for clarity, ```@DateFormat``` with no parameters gets us ISO8601 parsing of a date. The clunkiness comes from when we want both ISO8601 and something else (```@DateFormat({"myformathere", ISO8601_LONG, ISO8601_SHORT, etc.})```) vs when it's always a default and we don't want it (```@DateFormat(formats={"myonlyformat"}, omitISO8601=true)```).